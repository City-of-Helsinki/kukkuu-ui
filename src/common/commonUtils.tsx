import React from 'react';

export const nlToParagraph = (text: string) => {
  return (text || '').split('\n\r').map((item, i) => <p key={i}>{item}</p>);
};
