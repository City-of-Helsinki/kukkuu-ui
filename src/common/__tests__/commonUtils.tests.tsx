import { nlToParagraph } from '../commonUtils';

it('converts CRLFs to brs', () => {
  const string = 'a\r\nb';
  expect(nlToParagraph(string)).toMatchSnapshot();
});

it('converts LFs to brs', () => {
  const string = 'a\nb';
  expect(nlToParagraph(string)).toMatchSnapshot();
});
