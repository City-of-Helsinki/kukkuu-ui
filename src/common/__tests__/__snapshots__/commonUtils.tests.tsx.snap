// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converts CRLFs to brs 1`] = `
Array [
  <p>
    a
  </p>,
  <p>
    b
  </p>,
]
`;

exports[`converts LFs to brs 1`] = `
Array [
  <p>
    a
  </p>,
  <p>
    b
  </p>,
]
`;
